### PR TITLE
update data before graphing

### PR DESCRIPTION
**Pull request by ac615223s5**
update cache in case they had a new submission recently or haven't been cached yet

### CODE REVIEW

**Comment by jtyliu** [comment]
Caching submissions is intensive and slow. 

Putting them for plotting the graphs would mean it would add several seconds if was a new submission.

Personally, I feel that leaving this alone and making the user themselves recache and update is more efficient. It leaves the graphing commands faster, and whenever there's slightly outdated data it shouldn't make a big difference in the graph anyways.

Edit: Although, #11 does bring up a good idea, automatic caching after a few days or weeks would silently keep the data up to date, without having the user needing to cache after noticing their data is out of date.

**Comment by ac615223s5** [comment]
k then it'll just cache when it's never been cached before

**Comment by jtyliu** [comment]
That's fine, but can you send some sort of message like `No submissions found for "user", caching now`

Also please remove `not_cached`

**Comment by ac615223s5** [comment]
sure

File: cogs/plot.py [hunk 1]
```diff
@@ -146,7 +141,9 @@ def calculate_points(points, fully_solved):
 
             submissions = q.all()
             if len(submissions) == 0:
-                not_cached.append(username)
+                await ctx.send(f"`{username}` does not have any cached submissions,caching now`")
```
Comment (jtyliu) [inline cogs/plot.py]:
```suggestion
                await ctx.send(f"`{username}` does not have any cached submissions, caching now`")
```

File: cogs/plot.py [hunk 2]
```diff
@@ -299,13 +292,13 @@ def to_points(problem):
...
 
         max_percentage = 0
-        not_cached = []
 
         for username in usernames:
             q = session.query(Submission_DB)\
                 .filter(Submission_DB._user == username)
             if q.count() == 0:
-                not_cached.append(username)
+                await ctx.send(f"`{username}` does not have any cached submissions,caching now`")
```
Comment (jtyliu) [inline cogs/plot.py]:
```suggestion
                await ctx.send(f"`{username}` does not have any cached submissions, caching now`")
```

File: cogs/plot.py [hunk 3]
```diff
@@ -72,12 +72,12 @@ async def solved(self, ctx, *usernames):
             return await ctx.send('Too many users given, max 10')
 
         total_data = {}
-        not_cached = []
         for username in usernames:
             q = session.query(Submission_DB)\
                 .filter(Submission_DB._user == username)
             if q.count() == 0:
-                not_cached.append(username)
+                await ctx.send(f"`{username}` does not have any cached submissions,caching now`")
```
Comment (jtyliu) [inline cogs/plot.py]:
```suggestion
                await ctx.send(f"`{username}` does not have any cached submissions, caching now`")
```

**Review by jacklee1792 — COMMENTED** [review]
The lines with the message `{username} does not have any cached...` have an unmatched backtick

**Comment by ac615223s5** [comment]
🐒

**Comment by jtyliu** [comment]
Thanks!

**Pull request merged**